### PR TITLE
Feature/srm gpio wrapper class

### PR DESCRIPTION
**Pull request by Vavat**
Create a generic wrapper for the GPIO pins. This is meant in the future to support EXTI and change callbacks.

### CODE REVIEW

File: Lib/Inc/CGpioWrapper.h [hunk 1]
```diff
@@ -0,0 +1,28 @@
+/*
```
Comment (Niuslar) [inline Lib/Inc/CGpioWrapper.h]: I think this should follow the doxygen style for the file to be included?
```   
/**
/* @file CGpioWrapper.h
/*
```

File: Lib/Inc/CGpioWrapper.h [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+#ifndef CGPIOWRAPPER_H_
+#define CGPIOWRAPPER_H_
+
+#include "gpio.h"
+
+class CGpioWrapper
+{
+public:
+    CGpioWrapper(GPIO_TypeDef *p_port, uint32_t pin);
```
Comment (Niuslar) [inline Lib/Inc/CGpioWrapper.h]: If you're using the GPIO_PIN_X format, the max value is 0xFFFF (all pins), so no need for 32 bits. ```uint16_t``` should be enough.

File: Lib/Src/CGpioWrapper.cpp [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+ */
+CGpioWrapper::CGpioWrapper(GPIO_TypeDef *p_port, uint32_t pin)
+    : mp_port(p_port),
+      m_pin(pin)
+{
+    if (mp_port == NULL)
+    {
+        Error_Handler();
+    }
+    if (!IS_GPIO_PIN(m_pin))
```
Comment (Niuslar) [inline Lib/Src/CGpioWrapper.cpp]: If you use ```uint16_t``` I think this becomes redundant. Here's the IS_GPIO_PIN macro, let me know if you agree. 
```
#define GPIO_PIN_MASK              (0x0000FFFFU) /* PIN mask for assert test */
#define IS_GPIO_PIN(__PIN__)        ((((uint32_t)(__PIN__) & GPIO_PIN_MASK) != 0x00U) &&\
                                     (((uint32_t)(__PIN__) & ~GPIO_PIN_MASK) == 0x00U))
```

**Review by Niuslar — APPROVED** [review]
Some very minor comments.

File: Lib/Inc/CGpioWrapper.h (hunk 2 above)
Comment (Vavat) [reply Lib/Inc/CGpioWrapper.h]: But the CubeMX defines pins as uint32_t, so if I use uint16_t the compiler is going to give warnings about mismatched variables and reduced scope.

File: Lib/Inc/CGpioWrapper.h (hunk 1 above)
Comment (Vavat) [reply Lib/Inc/CGpioWrapper.h]: I did not know that. Cool.

File: Lib/Src/CGpioWrapper.cpp (hunk 3 above)
Comment (Niuslar) [reply Lib/Src/CGpioWrapper.cpp]: Just realised it's still useful to check the pin is not 0x00.

File: Lib/Inc/CGpioWrapper.h (hunk 2 above)
Comment (Vavat) [reply Lib/Inc/CGpioWrapper.h]: Turns out I was wrong that CubeMX defines them as 32 bit unsigned int. they are just #define values. I will change it.

File: Lib/Src/CGpioWrapper.cpp (hunk 3 above)
Comment (Vavat) [reply Lib/Src/CGpioWrapper.cpp]: It also checks that only one bit is set in a mask.

File: .cproject [hunk 4]
```diff
@@ -237,4 +238,5 @@
 		</scannerConfigBuildInfo>
 	</storageModule>
 	<storageModule moduleId="org.eclipse.cdt.make.core.buildtargets"/>
+	<storageModule moduleId="refreshScope"/>
```
Comment (Niuslar) [inline .cproject:241]: I think our .cproject files will always be different, perhaps it is better to ignore them and add a "settings setup" section in the ReadMe. (Add Lib to the include path, untick exclude from build, add support for floating point in printf, etc).

File: .project [hunk 5]
```diff
@@ -1,6 +1,6 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <projectDescription>
-	<name>Multichannel temperature sensor CPP</name>
+	<name>Multichannel temperature sensor</name>
```
Comment (Niuslar) [inline .project:3]: I think this one should be safe to ignore as well.

File: .cproject (hunk 4 above)
Comment (Vavat) [reply .cproject:241]: Don't know. Perhaps worth researching what .cproject file does.

Comment (Niuslar) [reply .cproject:241]: .project and .cproject are the IDE configuration files. .cproject is specific to the CDT (C/C++ plugin) and according to t[his link](https://www.eclipse.org/forums/index.php/t/174633/) 
> This defines things that are specific to C/C++ projects such as include
paths, library paths, and build commands.

**Pull request merged**